### ISSUE TITLE
Cell.Value = "Infinity"

### ISSUE THREAD

**Issue by ddrogahn-venga** [ClosedXML/ClosedXML#1676]
**Do you want to request a *feature* or report a *bug*?**
- [x] Bug
- [ ] Feature
- [ ] Question

**Did you test against the latest CI build?**
- [ ] Yes
- [x] No

**Version of ClosedXML**
0.95.4.0 (nuget.org)

**What is the current behavior?**
Cell().Value = seems to parse "Infinity" and "-Infinity" as a number, but opening the file in Excel gives a warning:
"We found a problem with some content ... do you want to try to recover?"
"Repaired Records: Cell information from /xl/worksheets/sheet1.xml part"

https://docs.microsoft.com/en-us/dotnet/api/documentformat.openxml.spreadsheet.cellvalues?view=openxml-2.8.1#DocumentFormat_OpenXml_Spreadsheet_CellValues_String
Specifically the CellType/Datatype "t" is set to "n" == Number
   <x:c r="A1" s="0" t="n">
    <x:v>Infinity</x:v>
   </x:c>

"NaN" does not trigger this problem.
The string for infinity may change with the culture? unless InvariantCulture is used.
https://stackoverflow.com/questions/18398690/how-should-i-parse-infinity-with-double
(I can't seem to code Decimal.Parse to take "Infinity" or "∞"...)

**What is the expected behavior or new feature?**
Excel doesn't have a representation for infinity: =-1*LOG10(0) returns #NUM!   =1/0 returns #DIV/0!
So "Infinity", "-Infinity" and localizations should be interpreted as a string.

**Is this a regression from the previous version?**
No

## Reproducibility

**Code to reproduce problem:**
```c#
public void Main()
{
	using (var wb = new XLWorkbook()) {
		IXLWorksheet ws = wb.Worksheets.Add("Sheet 1");
		ws.Cell(1, 1).Value = "Infinity";
		wb.SaveAs("Infinity.xlsx");
	}
}
```

- [x] I attached a sample spreadsheet.  (You can drag files on to this issue)
attached
[Infinity.xlsx](https://github.com/ClosedXML/ClosedXML/files/6626502/Infinity.xlsx)

**Comment by igitur** [comment]
That's a fairly rare case. But why don't you attempt a PR to fix this?
Thanks

**Comment by sylvaincaillot** [comment]
Same problem, I am just trying to store "Infinity" which is the name of a product in my Excel file and it gives me an error "We found a problem with some content ... do you want to try to recover?"

How can I proceed to save that particular text? Is there any other  keywords I should be looking at?

**Comment by igitur** [comment]
You can use `cell.Value = "'Infinity"` or `cell.SetValue("Infinity")`

**Comment by sylvaincaillot** [comment]
Perfect. I have replaced my `worksheet.Cell(currentRow, 3).Value = vr.Name`  by `worksheet.Cell(currentRow, 3).SetValue(vr.Name);`.
It works nice.

**Comment by Listbd** [comment]
Thank you @igitur !!
Spent a few hours tracking it down to that particular value from a large set of data I was trying to use, and then just stared blankly at the screen when I realized it was just that word.

**Comment by jahav** [comment]
I think this has been fixed in [0.100.0](https://github.com/ClosedXML/ClosedXML/releases/tag/0.100.0) (heading _Cell data type is no longer guessed_).

Anyway, the setter no longer sets the value to infinity using provided repro, nor is it possible to set infinity explicitely (`ws.Cell(1, 1).Value = double.PositiveInfinity` will throw exception, because infinity is not a valid `XLCellValue`).
![image](https://github.com/user-attachments/assets/5ba083c4-fd6c-4cb8-935c-7d96a3f896b6)
